### PR TITLE
filter specialization for Strings (#32460)

### PR DESCRIPTION
**Pull request by jsulmont**

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]

File: base/strings/string.jl [hunk 1]
```diff
@@ -347,3 +347,16 @@ function repeat(c::Char, r::Integer)
     end
     return s
 end
+
+function Base.filter(f, s::String)
```
Comment (KristofferC) [inline base/strings/string.jl]: Can remove the Base prefixes since we are in Base.

File: test/strings/basic.jl [hunk 2]
```diff
@@ -113,7 +113,10 @@ end
     @test endswith(z, z)
 end
 
-@test filter(x -> x ∈ ['f', 'o'], "foobar") == "foo"
+@testset "filter specialization on String issue #32460" begin
+     @test filter(x -> x ∈ ['f', 'o'], "foobar") == "foo"
```
Comment (KristofferC) [inline test/strings/basic.jl]: Perhaps take the opportunity to add a few "harder" tests with the string containing a few Greek letters for example.

Comment (jsulmont) [reply test/strings/basic.jl]: @KristofferC I think that one should be ready to go!

**Comment by KristofferC** [comment]
I added a commit to fix the trailing whitespace issues: https://build.julialang.org/#/builders/150/builds/206.

**Comment by jsulmont** [comment]
sorry 😕

**Comment by StefanKarpinski** [comment]
No worries, thanks for leaving your branch open to commits from maintainers so we can help :)

**Pull request merged**